addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/launch-product/launch-product.js

```diff
@@ -90,7 +91,9 @@ class LaunchProduct extends StepBase {
 
     // Read input params specified in the environment type configuration
     // The params may include variable expressions, resolve the expressions by using the resolveVars
-    const resolvedInputParams = await this.resolveVarExpressions(envTypeConfig.params, resolvedVars);
+    const resolvedInputParamsRaw = await this.resolveVarExpressions(envTypeConfig.params, resolvedVars);
+    // Additional layer to check the namespace is valid and unique
```

Comment:
I am guessing that namespace might have become static after going through `resolveVarExpressions` method. We should probably add a comment to explain why this additional layer of check is required because just before `resolveVarExpressions` we do have a namespace ending with current date.

Reply:
Added comment